rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/ApplicationResource.java

```diff
@@ -620,11 +620,21 @@ private void checkLocationsAreValid(ApplicationSpec applicationSpec) {
 
     @Override
     public List<EntitySummary> getDescendants(String application, String typeRegex) {
+        Entity entity =brooklyn().getApplication(application);
+        if (entity != null && !Entitlements.isEntitled(mgmt().getEntitlementManager(), Entitlements.SEE_ENTITY, entity)) {
+            throw WebResourceUtils.forbidden("User '%s' is not authorized to see the entity '%s'",
+                    Entitlements.getEntitlementContext().user(), entity);
+        }
         return EntityTransformer.entitySummaries(brooklyn().descendantsOfType(application, application, typeRegex), ui.getBaseUriBuilder());
     }
 
     @Override
     public Map<String, Object> getDescendantsSensor(String application, String sensor, String typeRegex) {
+        Entity entity =brooklyn().getApplication(application);
+        if (entity != null && !Entitlements.isEntitled(mgmt().getEntitlementManager(), Entitlements.SEE_ENTITY, entity)) {
+            throw WebResourceUtils.forbidden("User '%s' is not authorized to see the entity '%s'",
```

Comment:
Message should say `User '%s' is not authorized to see the value of sensor '%s' for descendants of entity '%s'"`
Please also check the parameters for the templated message